tests/sentry/api/serializers/test_snuba.py

```diff
@@ -51,3 +55,51 @@ def test_non_rollup_buckets(self):
             rollup=rollup,
             zerofilled_buckets=[2, 3, 4, 5],
         )
+
+
+class SnubaTSResultSerializerTest(TestCase):
+    def run_test(self, start, values, rollup, expected, max_buckets=None):
+        start = start.replace(tzinfo=pytz.utc, microsecond=0, second=0)
+        end = start + (rollup * len(values))
+
+        data = []
+        for i, value in enumerate(values):
+            row = {}
+            row["time"] = int(to_timestamp(start + (rollup * i)))
+            if value is not None:
+                row["count"] = value
+            data.append(row)
+
+        result = SnubaTSResult({"data": data}, start, end, int(rollup.total_seconds()))
+        serializer = SnubaTSResultSerializer(
+            self.organization, None, self.user, max_buckets=max_buckets
+        )
+        assert [item[1][0] for item in serializer.serialize(result)["data"]] == expected
+
+    def test(self):
+        self.run_test(
+            before_now(days=1),
+            [None, 2, 1, None],
+            timedelta(minutes=1),
+            [{"count": 0}, {"count": 2}, {"count": 1}, {"count": 0}],
+        )
+
+    def test_max_buckets(self):
+        self.run_test(
+            before_now(days=1),
+            [None, 2, 1, None],
+            timedelta(minutes=1),
+            [{"max": 2, "avg": 1.0, "min": 0}, {"max": 1, "avg": 0.5, "min": 0}],
```

Comment:
Will the response values always have this shape when maxBuckets is used? Or will it sometimes be the simple count shape?

Reply:
So if we don't exceed `maxBuckets` then it'll be the simple format. This is because we don't really want to take these aggregates when we have the exact values to work with. When I discussed that with the FE engineers on my team they seemed fine with it.

If we don't like this we could:
 - include some value that people can check to see which type of data they're receiving
 - always return the aggregate format, and just use a bucket size of 1 when we're below `maxBuckets`

Reply:
Sending two requests for different timeframes but otherwise using the same parameters and getting very different results doesn't feel like an ideal interface as you always need to handle both modes and it isn't transparent what causes the output format to change.

A hint on what format has been sent could help folks that aren't our UI that might be consuming these APIs. But those folks would be unlikely to be using this new parameter so the data would likely go unused. If the UI code is ok with sniff testing the results I don't want to be a blocker or cause more work.

Reply:
I'm also ok with just always showing the aggregate format when `maxBuckets` is used. Then we don't need to handle two modes. Not sure whether @scttcper @taylangocmen have any thoughts here after seeing these options?

Reply:
I think one thing is that we might want to show a slightly different ui if we've aggregated this data. Like just give the users a heads-up that they're not seeing exact values. So to support that, the best options are either the sniff test, or returning a specific hint.